### PR TITLE
JNG-4827 validate that fields with abstarct entity type are invalid

### PR DESCRIPTION
**Pull request by Madmat8**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4827" title="JNG-4827" target="_blank">StoryJNG-4827</a>  Composite relations targeting abstract entities shall be invalid
  </td></table>
  <br />

### CODE REVIEW

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 1]
```diff
@@ -926,6 +927,23 @@ class JslDslValidator extends AbstractJslDslValidator {
...
         }
     }
+    
+    @Check
+    def checkAbstractComposition(EntityMemberDeclaration member) {
+        if (member instanceof EntityFieldDeclaration) {
+            if (member.referenceType instanceof EntityDeclaration){
+                val reference = member.referenceType as EntityDeclaration
+	            if (reference.isIsAbstract()) {
+	            	error("You cannot use entity named: '" + (member.referenceType as EntityDeclaration).name + "'as a field type, because it is abstract.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: I suggest removing `:` and adding space to `"'as`

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 2]
```diff
@@ -926,6 +927,23 @@ class JslDslValidator extends AbstractJslDslValidator {
             }
         }
     }
+    
+    @Check
+    def checkAbstractComposition(EntityMemberDeclaration member) {
+        if (member instanceof EntityFieldDeclaration) {
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: Wouldn't it be more efficient if the definition parameter would be `EntityFieldDeclaration`?

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 1 above)
Comment (Madmat8) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: [406563b](https://github.com/BlackBeltTechnology/judo-meta-jsl/pull/149/commits/406563b2452540ccfd130360ae0e84b61257e2bb)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 2 above)
Comment (Madmat8) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: [406563b](https://github.com/BlackBeltTechnology/judo-meta-jsl/pull/149/commits/406563b2452540ccfd130360ae0e84b61257e2bb)

**Review by noherczeg — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**